### PR TITLE
creating project juction tables with editing and deleteing fileds

### LINKED ISSUE #100: 37-connecting-project-table-with-junction-tables

### PR DESCRIPTION
**Pull request by Moamin-AbuEwaida**
creating the code for the junction tables of Projects table with the edit,delete features. also added the code for editing projects table entries

### CODE REVIEW

File: api/controllers/project.js [hunk 1]
```diff
@@ -45,6 +44,157 @@ const projectController = {
...
+      });
+    } catch (error) {
+      console.log(error);
+      res.status(500).json({ message: error.message });
+    }
+  },
+  deleteClass: async (req, res) => {
+    try {
+      const { classId, projectId } = req.params;
+      await userManager.deleteProjectClass({
```
Comment (rayanejsilva) [inline api/controllers/project.js]: Here is projectManager instead of userManager.

File: api/controllers/project.js [hunk 2]
```diff
@@ -45,6 +44,157 @@ const projectController = {
...
```
Comment (rayanejsilva) [inline api/controllers/project.js]: Here is projectManager instead of userManager.

File: api/controllers/project.js [hunk 3]
```diff
@@ -45,6 +44,157 @@ const projectController = {
...
+      });
+    } catch (error) {
+      console.log(error);
+      res.status(500).json({ message: error.message });
+    }
+  },
+  deleteUser: async (req, res) => {
+    try {
+      const { userId, projectId } = req.params;
+      await userManager.deleteUserProject({
```
Comment (rayanejsilva) [inline api/controllers/project.js]: Here is projectManager instead of userManager.

File: api/controllers/project.js [hunk 4]
```diff
@@ -45,6 +44,157 @@ const projectController = {
...
+      });
+      res.status(200).json({
+        message: `Class with id ${classId} was successfully deleted!`,
+      });
+    } catch (error) {
+      console.log(error);
+      res.status(500).json({ message: error.message });
+    }
+  },
+  deleteClass: async (req, res) => {
```
Comment (rayanejsilva) [inline api/controllers/project.js]: deleteClass is duplicated.

File: api/routes/project.js [hunk 5]
```diff
@@ -1,24 +1,37 @@
...
+const projectManager = require("../business-logic/project");
 
 const projectRoutes = express.Router();
 
 projectRoutes.get("/", projectController.get);
 projectRoutes.post("/", projectController.post);
-projectRoutes.get("/:projectId", projectController.getProjectById)
-
+projectRoutes.get("/:projectId", projectController.getProjectById);
+projectRoutes.put("/:projectId/project", projectController.putProject);
```
Comment (rayanejsilva) [inline api/routes/project.js]: I would remove project after /:projectId. **_Line 10_**

File: api/routes/project.js [hunk 6]
```diff
@@ -1,24 +1,37 @@
...
+projectRoutes.post("/:projectId/skill", projectController.postSkill);
+projectRoutes.post("/:projectId/class", projectController.postClass);
+projectRoutes.post("/:projectId/user", projectController.postUser);
+projectRoutes.delete("/:projectId/project", projectController.deleteProject);
```
Comment (rayanejsilva) [inline api/routes/project.js]: I would remove project after /:projectId. **_Line 14_**

Comment (Moamin-AbuEwaida) [reply api/routes/project.js]: :eyes:

**Review by rayanejsilva — APPROVED** [review]
LGTM!

**Pull request merged**